vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -199,47 +200,73 @@ public Retrofit getRetrofit() {
 
+    /**
+     * @return the current VimeoAccount, or null if it hasn't been set yet
+     */
+    @Nullable
     public VimeoAccount getVimeoAccount() {
-        if (mVimeoAccount == null) {
-            throw new AssertionError("Account should never be null");
-        }
+        return VimeoAccount.copy(mVimeoAccount);
+    }
 
-        return mVimeoAccount;
+    /**
+     * @return true if the current {@link VimeoAccount} is not null and has an auth token
+     */
+    public boolean isAuthenticated() {
+        return mVimeoAccount != null && mVimeoAccount.isAuthenticated();
     }
 
+    /**
+     * Sets the current {@link VimeoAccount} on this client. If the provided account is null and
+     * an access token exists on the {@link Configuration}, then a new account will be created
+     * using that access token.
+     *
+     * @param vimeoAccount the account to set on this client
+     */
     public void setVimeoAccount(@Nullable VimeoAccount vimeoAccount) {
         if (vimeoAccount == null) {
-            // If the provided account was null but we have an access token, persist the vimeo account with
-            // just a token in it. Otherwise we'll want to leave the persisted account as null.
-            vimeoAccount = new VimeoAccount(mConfiguration.accessToken);
             if (mConfiguration.accessToken != null) {
-                mConfiguration.saveAccount(vimeoAccount, null);
+                // If the provided account was null but we have an access token, persist the vimeo account with
+                // just a token in it. Otherwise we'll want to leave the persisted account as null.
+                vimeoAccount = new VimeoAccount(mConfiguration.accessToken);
+                mConfiguration.saveNonUserAccount(vimeoAccount);
             }
         }
 
         mVimeoAccount = vimeoAccount;
     }
 
     /**
-     * @deprecated use {@link #saveAccount(VimeoAccount, String)} instead
-     * <p/>
-     * We find no use in storing the password when you can persist the {@link VimeoAccount} across
-     * application sessions.
+     * Sets the {@link #mVimeoAccount} field as well as triggering the saveAuthenticatedUserAccount
+     * event for the account store.
+     *
+     * @param vimeoAccount The account to save - this should be provided; if it is not, then one will
+     *                     be constructed using the access token set on the {@link Configuration}.
+     * @param accountName  accountName should be provided if the account is for an authenticated user. It may
+     *                     be null if the account represents a client credentials grant.
      */
-    @Deprecated
-    public void saveAccount(@Nullable VimeoAccount vimeoAccount, String email, String password) {
-        saveAccount(vimeoAccount, email);
+    public void saveAuthenticatedUserAccount(@Nullable VimeoAccount vimeoAccount,
+                                             @NotNull String accountName) {
+        setVimeoAccount(vimeoAccount);
```

Comment:
Similar comment here, it seems strange to be about to "save an authenticated user account that is null". What does saving a null account mean? Can this API be clarified so the method name(s) and implementation(s) are crystal clear?